interactive/interactive.go

```diff
@@ -35,6 +38,27 @@ func OpenInBrowser(url string) error {
 	return err
 }
 
+// HostOSPlatform returns the host's OS platform akin to `runtime.GOOS`, with
+// added awareness of Windows Subsystem for Linux (WSL) 2 environments.
+// The possible values are the same as `runtime.GOOS`, plus "WSL2".
+func HostOSPlatform() string {
+	if wsl2() {
+		return "WSL2"
+	}
+	return runtime.GOOS
+}
+
+func wsl2() bool {
+	if runtime.GOOS != "linux" {
+		return false
+	}
+	version, err := os.ReadFile("/proc/version")
+	if err != nil {
+		return false
+	}
+	return bytes.Contains(version, []byte("WSL2"))
+}
```

Comment:
Repeating old comment here for visibility after some updates: I would love to move these two functions into https://github.com/efficientgo/core. If maintainers agree, I'll open a PR there and update this one.

Reply:
Hm... We could consider that. Feel free to add TODO with this, but I ideally we test it's usability (how useful it is for others) in e2e first.